src/ol_infrastructure/applications/edxapp/k8s_resources.py

```diff
@@ -0,0 +1,960 @@
+        api_version="vpcresources.k8s.aws/v1beta1",
+        kind="SecurityGroupPolicy",
+        metadata=kubernetes.meta.v1.ObjectMetaArgs(
+            name=f"{env_name}-edxapp-sg-policy",
+            namespace=namespace,
+            labels=k8s_global_labels,
+        ),
+        spec={
+            "podSelector": {
+                "matchLabels": {
+                    "ol.mit.edu/pod-security-group": edxapp_k8s_app_security_group.id,
+                }
+            },
+            "securityGroups": {"groupIds": [edxapp_k8s_app_security_group.id]},
+        },
+        opts=pulumi.ResourceOptions(depends_on=[edxapp_k8s_app_security_group]),
+    )
+
+    secrets = create_k8s_secrets(
+        edxapp_cache=edxapp_cache,
+        edxapp_config=edxapp_config,
+        edxapp_db=edxapp_db,
+        k8s_global_labels=k8s_global_labels,
+        mongodb_atlas_stack=mongodb_atlas_stack,
+        namespace=namespace,
+        stack_info=stack_info,
+        vault_k8s_resources=vault_k8s_resources,
+    )
+
+    configmaps = create_k8s_configmaps(
+        stack_info=stack_info,
+        namespace=namespace,
+        k8s_global_labels=k8s_global_labels,
+        edxapp_config=edxapp_config,
+        edxapp_cache=edxapp_cache,
+        notes_stack=notes_stack,
+        opensearch_hostname=opensearch_hostname,
+    )
+
+    # Lookup environment vars needed for the deployment
+    if "EDXAPP_DOCKER_IMAGE_DIGEST" not in os.environ:
+        msg = "Environment variable EDXAPP_DOCKER_IMAGE_DIGEST is not set. "
+        raise OSError(msg)
+    EDXAPP_DOCKER_IMAGE_DIGEST = os.environ["EDXAPP_DOCKER_IMAGE_DIGEST"]
+
+    OPENEDX_RELEASE: OpenEdxSupportedRelease = os.environ.get(
+        "OPENEDX_RELEASE", "master"
+    )
+
+    ############################################
+    # Setup an init container that downloads and extracts the staticfiles
+    ############################################
+    edxapp_image = f"610119931565.dkr.ecr.us-east-1.amazonaws.com/dockerhub/mitodl/edxapp@{EDXAPP_DOCKER_IMAGE_DIGEST}"
+    production_staticfiles_archive_name = (
+        f"staticfiles-production-{EDXAPP_DOCKER_IMAGE_DIGEST}.tar.gz"
+    )
+    nonprod_staticfiles_archive_name = (
+        f"staticfiles-nonprod-{EDXAPP_DOCKER_IMAGE_DIGEST}.tar.gz"
+    )
+    production_staticfiles_url = f"https://ol-eng-artifacts.s3.amazonaws.com/edx-staticfiles/{stack_info.env_prefix}/{OPENEDX_RELEASE}/{production_staticfiles_archive_name}"
+    nonprod_staticfiles_url = f"https://ol-eng-artifacts.s3.amazonaws.com/edx-staticfiles/{stack_info.env_prefix}/{OPENEDX_RELEASE}/{nonprod_staticfiles_archive_name}"
+
+    prod_dl_command = f"wget {production_staticfiles_url} -O /tmp/prod.tar.gz && tar -xf /tmp/prod.tar.gz --strip-components 2 -C /openedx/staticfiles"
+    nonprod_dl_command = f"wget {nonprod_staticfiles_url} -O /tmp/nonprod.tar.gz && tar -xf /tmp/nonprod.tar.gz --strip-components 2 -C /openedx/staticfiles"
```

Comment:
I would like to get rid of the external staticfiles management as it is no longer necessary with the ProctorTrack integration and it adds a lot of cognitive and build overhead.